src/vs/workbench/services/backup/node/backupFileService.ts

```diff
@@ -147,6 +158,10 @@ export class BackupFileService implements IBackupFileService {
 				return void 0; // return early if backup version id matches requested one
 			}
 
+			if (resource.scheme === 'file') {
```

Comment:
@Tyriar so you say that you do this only for files because untitled do not have this issue? not sure about that, what about someone that starts "`code <file name>`" where the file path does not exist and we show an untitled file with associated file path? I think we will have to store the path for untitled too in that case. Maybe untitled files should be stored differently on disk compared to today: instead of using their name (`Untitled-1`, `Untitled-2`), hash the associated file path and if there is none, hash the name.